src/ext/isearch.lisp

```diff
@@ -347,30 +347,42 @@
     (when (string= "" *isearch-string*)
       (setq *isearch-string* (isearch-default-string)))
     (setf (variable-value 'isearch-prev-last :buffer) nil)
-    (cond ((variable-value 'isearch-next-last :buffer)
-           (setf (variable-value 'isearch-next-last :buffer) nil)
-           (with-point ((p (current-point)))
-             (buffer-start p)
-             (when (funcall *isearch-search-forward-function* p *isearch-string*)
-               (move-point (current-point) p))))
-          ((not (funcall *isearch-search-forward-function* (current-point) *isearch-string*))
-           (setf (variable-value 'isearch-next-last :buffer) t)))
-    (isearch-update-display)))
+    (prog1
+        (cond ((variable-value 'isearch-next-last :buffer)
+               (setf (variable-value 'isearch-next-last :buffer) nil)
+               (with-point ((p (current-point)))
+                 (buffer-start p)
+                 (if (funcall *isearch-search-forward-function* p *isearch-string*)
+                     (progn
+                       (move-point (current-point) p)
+                       t)
+                     nil)))
+              ((not (funcall *isearch-search-forward-function* (current-point) *isearch-string*))
+               (setf (variable-value 'isearch-next-last :buffer) t)
+               nil)
+              (t))
+      (isearch-update-display))))
 
 (define-command isearch-prev () ()
   (when (boundp '*isearch-string*)
     (when (string= "" *isearch-string*)
       (setq *isearch-string* (isearch-default-string)))
     (setf (variable-value 'isearch-next-last :buffer) nil)
-    (cond ((variable-value 'isearch-prev-last :buffer)
-           (setf (variable-value 'isearch-prev-last :buffer) nil)
-           (with-point ((p (current-point)))
-             (buffer-end p)
-             (when (funcall *isearch-search-backward-function* p *isearch-string*)
-               (move-point (current-point) p))))
-          ((not (funcall *isearch-search-backward-function* (current-point) *isearch-string*))
-           (setf (variable-value 'isearch-prev-last :buffer) t)))
-    (isearch-update-display)))
+    (prog1
+        (cond ((variable-value 'isearch-prev-last :buffer)
+               (setf (variable-value 'isearch-prev-last :buffer) nil)
+               (with-point ((p (current-point)))
+                 (buffer-end p)
+                 (if (funcall *isearch-search-backward-function* p *isearch-string*)
+                     (progn
+                       (move-point (current-point) p)
+                       t)
+                     nil)))
+              ((not (funcall *isearch-search-backward-function* (current-point) *isearch-string*))
+               (setf (variable-value 'isearch-prev-last :buffer) t)
+               nil)
+              (t))
```

Comment:
This returns T. Is this intended?

Reply:
Yep. I changed this to return a boolean value as I did to `isearch-next`.

Reply:
Oh, I understand now 👍